source/features/clean-repo-sidebar.tsx

```diff
@@ -31,8 +31,21 @@ async function cleanReleases(): Promise<void> {
 
 async function hideEmptyPackages(): Promise<void> {
 	const packagesCounter = await elementReady('.Layout-sidebar .BorderGrid-cell a[href*="/packages?"] .Counter', {waitForChildren: false});
-	if (packagesCounter && packagesCounter.textContent === '0') {
-		packagesCounter.closest('.BorderGrid-row')!.hidden = true;
+	if (!packagesCounter || packagesCounter.textContent !== '0') {
+		return;
+	}
+
+	packagesCounter.closest('.BorderGrid-row')!.hidden = true;
+
+	// Github seems to fetch internal packages separately, so if a repo only has internal packages, this makes the panel visible after they've loaded
+	const updatedPackagesCounter = await elementReady('.Layout-sidebar .BorderGrid-cell a[href*="/packages?"] .Counter:not([hidden="hidden"])', {
+		waitForChildren: false,
+		stopOnDomReady: false,
+		timeout: 10 * 1000,
```

Comment:
The problem with this, other than the repeated code, is that it affects everyone else negatively. `elementReady` is relatively heavy because it queries the DOM up to 60 times per second, for 10 seconds in this case.

What you want is to target the "deferred" placeholder element that GitHub probably has, and wait for it to trigger the load event.

I think the code would look like:

```js
const el = await elementReady('deferred-content[src*=packages]');
await oneEvent(el, 'load')
hidden = false
```

Reply:
By the way I don't know what the DOM in that case looks like so it might be good to click "View source" on your repo page and copy the HTML from there, which should show that "deferred content" element (I don't remember the exact name)

Reply:
Is it `include-fragment`? https://github.com/github/include-fragment-element

Reply:
Correct